pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,72 +77,79 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
-	// to be added to the total amount posted to prometheus
-	var cancelledCount, completeCount, errorCount, newCount, pendingCount, runningCount int
-	reasons := map[string]int{}
+	// collectBuild will return counts for the build's phase/reason tuple,
+	// and counts for these tuples be added to the total amount posted to prometheus
+	counts := map[string]map[string]int{
```

Comment:
I think defining a `type foo struct { phase, reason string }` and a single-level `map[foo]int` would be clearer

Reply:
agreed / like that ... part of next push